src/pages/Episode/Episode.js

```diff
@@ -1,39 +1,101 @@
 import React, { Component } from "react";
 
+import { EPISODE, URL } from "../../constants/routes";
 import Layout from "../../components/Layout";
-// import CharacterCard from "../../components/CharacterCard";
+import CharacterCard from "../../components/CharacterCard";
+
+const axios = require("axios");
 
 class Episode extends Component {
   constructor(props) {
     super(props);
 
-    this.state = {};
-    // episode: null,
-    // characters: [],
-    // hasLoaded: false,
-    // hasError: false,
-    // errorMessage: null,
+    this.state = {
+      name: null,
+      episode: null,
+      airDate: null,
+      characters: [],
+      hasLoaded: false,
+      hasError: false,
+      errorMessage: null,
+    };
+
+    this.loadSingleEpisode = this.loadSingleEpisode.bind(this);
+  }
+
+  async componentDidMount() {
+    await this.loadSingleEpisode();
+  }
+
+  async loadSingleEpisode() {
+    try {
+      const { location } = this.props;
+      const id = location.pathname.replace("/episode/", "");
+      const response = await axios.get(`${URL}${EPISODE}/${id}`);
+      const data = response.data;
+      const { name, episode, air_date: airDate, characters } = data;
+      const pjs = characters.map((pj) => axios.get(pj));
+      const charactersArr = [];
+      // eslint-disable-next-line compat/compat
+      await Promise.all(pjs).then((result) =>
+        result.forEach((r) => charactersArr.push(r.data)),
+      );
```

Comment:
This is a very well done code. However, the requests could have been made differently if you had had a separate file to make requests such as a request to get an specific episode using the episodeId.  if you had that you could have an alternative code that would be something along the lines of this:

`const { data } = await getEpisode(episodeId);
      const promises = data.characters.map((character) => axios.get(character));
   
      const charactersResponse = await Promise.all(promises);

      const characters = charactersResponse.map((character) => character.data); `

This way, the code could be simplified a little bit more.